src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackEncoder.cs

```diff
@@ -199,6 +38,43 @@ public static bool EncodeIndexedHeaderField(int index, Span<byte> destination, o
             return false;
         }
 
+        /// <summary>Encodes the status code of a response to the :status field.</summary>
+        public static bool EncodeStatusHeader(int statusCode, Span<byte> destination, out int bytesWritten)
```

Comment:
I would merge this idea into `StatusCodes.ToStatusBytes()`. You can have the static arrays there store the pre-encoded version of the full header.

Reply:
I'll leave that for now. It can be a future optimization.